src/v/v8_engine/instance.cc

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "v8_engine/instance.h"
+
+#include "utils/file_io.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/temporary_buffer.hh>
+#include <seastar/util/later.hh>
+
+#include <utility>
+
+namespace v8_engine {
+
+instance::instance(size_t max_heap_size_in_bytes) {
```

Comment:
You can initialize in line:
```
instance::instance(size_t max_heap_size_in_bytes) : _create_params.array_buffer....(std::make_shared<>(.....)), ..... {}
```

Reply:
This way you can also mark these variables as const if you wanted to

Reply:
Done(delete create_params from instance)